pwnlib/elf/pltgot.py

```diff
@@ -0,0 +1,291 @@
+"""Utilities to parse the PLT and GOT of an ELF file.
+
+Style note: these utilities should work with plain elftools ELFFiles.
+"""
+
+from elftools.elf.constants import SHN_INDICES
+from elftools.elf.relocation import RelocationSection
+from pwnlib.util import packing
+from pwnlib.log import getLogger
+
+log = getLogger(__name__)
+
+def read_got(elf):
+    """ Read GOT symbols and corresponding program locations from the binary. """
+
+    got = {}
+
+    # TODO: In reality, the canonical relocation list is the one linked from
+    # the DYNAMIC segment; if we parse that, then we can get GOT even if the
+    # section headers are missing or obfuscated away.
+    for sect in elf.iter_sections():
+        if not isinstance(sect, RelocationSection):
```

Comment:
This wil fail for Android binaries built with the NDK.  See the related comment below.